### PR TITLE
Serialize memory_format

### PR DESCRIPTION
**Pull request by angelayi**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #81332

### CODE REVIEW

**Comment by angelayi** [comment]
@angelayi has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D37789826).

File: torch/csrc/MemoryFormat.cpp [hunk 1]
```diff
@@ -28,6 +28,17 @@ PyObject* THPMemoryFormat_repr(THPMemoryFormat* self) {
   return THPUtils_packString(self->name);
 }
 
+PyObject* THPMemoryFormat_reduce(PyObject* _self, PyObject* noargs) {
+  auto self = (THPMemoryFormat*)_self;
```
Comment (qihqi) [inline torch/csrc/MemoryFormat.cpp:32]: nit: auto* self

**Review by qihqi — APPROVED** [review]
Gogogo thanks!

**Comment by angelayi** [comment]
@angelayi has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D37789826).

**Pull request closed without merging**